models/customer.js

```diff
@@ -0,0 +1,80 @@
+var app = require("../app");
+var db = app.get("db");
+
+var Customer = function(customer) {
+  this.id = customer.id
+  this.name = customer.name
+  this.registered_at = customer.registered_at
+  this.address = customer.address
+  this.city = customer.city
+  this.state = customer.state
+  this.postal_code = customer.postal_code
+  this.phone = customer.phone
+  this.account_credit = customer.account_credit
+  this.due_date = customer.due_date
+  this.checkout_date = customer.checkout_date
+  this.title = customer.title
+}
+module.exports = Customer;
+
+var Movie = require("../models/movie");
+
+// takes on parameter(callback)-then run db.accounts.find
+Customer.all = function(callback) {
+  // then run db.accounts.find(no specific id or column - just another callback)
+  db.customers.find(function(error, customers) {
+    if(error || !customers) {
+      // handling any error
+      callback(error || new Error("Could not retrieve customers"), undefined);
+    } else {
+      // saying there was no error, accounts is an array and we map it
+      callback(null, customers.map(function(customer) {
+        // and return to a new instance of the account with id
+        return new Customer(customer);
```

Comment:
curious- what is the purpose of creating all the new instances of Customer? why not just return `customers` (without mapping)?